code-postprocessing/cocopp/ppfig.py

```diff
@@ -545,25 +545,29 @@ def plotUnifLogXMarkers(x, y, nbperdecade, logscale=False, **kwargs):
 
     This function only works with monotonous graph.
     """
-    res = plt.plot(x, y, **kwargs)  # shouldn't this be done in the calling code?
+    line_args = kwargs.copy()
+    line_args['marker'] = ''
+    line_args['label'] = ''
+    res = plt.plot(x, y, **line_args)  # shouldn't this be done in the calling code?
 
     if 'marker' in kwargs and len(x) > 0:
         # x2, y2 = downsample(x, y)
         x2, y2 = marker_positions(x, y, nbperdecade, 19, plt.axis(),
                                   np.log10 if logscale else None)
-        res2 = plt.plot(x2, y2)
-        for i in res2:
-            i.update_from(res[0])  # copy all attributes of res
-        plt.setp(res2, linestyle='', label='')
+        marker_args = kwargs.copy()
+        marker_args['drawstyle'] = 'default'
+        marker_args['linestyle'] = ''
+        marker_args['label'] = ''
+        res2 = plt.plot(x2, y2, **marker_args)
```

Comment:
I noticed that in the new code the `drawstyle` is set to 'default' (for `res2`) but in the old code it wasn't. Similarly below `linestyle` and `marker` are set to '' (for `res3`). Was this intended?

Reply:
Indeed, all intended.

The first plot is to draw the line, the second is to place the markers and the third is to add the label. In `kwargs`, `linestyle`, `marker` and `label` are provided to style each of these plot elements, respectively. For the first plot, `linestyle` matters but `marker` and `label` must be unset. For the second plot, `marker` matters but `linestyle` and `label` must be unset. For the third plot, `label` matters and since it's a plot of `NaN` vs `NaN`, `linestyle` or `marker` are supposed to be ineffectual. However, since making a plot invisible by passing `NaN` for coordinates is relying on exceptional handling of this special value internal to `matplotlib` and the same effect could be accomplished by unsetting `linestyle` and `marker` in a more self-explanatory fashion, I thought I'd add the additional code in. After all, passing empty lists as coordinates also worked until recently. Better to not have to keep in mind or rely on `matplotlib`'s handling of `NaN` values.

Setting `drawstyle` to `default` for the marker plot _is_ the actual fix. When set to `steps`, `matplotlib` places extra points so that the plot looks like a staircase. That may work with the line but surely not with the way post-processing uses markers.

Reply:
Thanks for the clarification. Then it's fine for me, it can be merged.